cpp/src/strings/char_types/char_types.cu

```diff
@@ -83,6 +83,107 @@ std::unique_ptr<column> all_characters_of_type(
   return results;
 }
 
+namespace {
+
+/**
+ * @brief Removes individual characters from a strings column based on character type.
+ *
+ * Types to remove are specified by `types_to_filter` OR
+ * types to not remove are specified by `types_to_keep`.
```

Comment:
The "OR" here seems to indicate this should maybe be two separate functions.

Reply:
I wanted to leave the door open for more flexible use cases in the future.